src/frontend/src/actions/index.ts

```diff
@@ -27,11 +27,20 @@ const mapStoreToProps = (state: globalState) => state;
 
 const mapDispatchToProps = (dispatch: Function) => {
 
-  async function asyncAction<T>(pr: Promise<T>) {
+  async function asyncAction<T>(pr: Promise<T>): Promise<T> {
     try {
       const result = await pr;
       return result;
     } catch (e) {
+      // in case of a conflict, override regular error handling
+      if (e instanceof ConflictError) {
+        dispatch({
+          type: appStateActions.conflictOccurred,
+          payload: e.results
+        });
+        await actions.dispatch.dialog.toggleResolveConflict();
+        return Promise.reject(e.message);
```

Comment:
throw e.message is equivalent (and we probably don't want to just throw a raw string).